app/services/bulk_update/recommendations_uploads/validate_trainee.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+module BulkUpdate
+  module RecommendationsUploads
+    # finds and validates a trainee scoped within provider, via trn/hesa/provider_trainee_id
+    class ValidateTrainee
+      def initialize(row:, provider:)
+        @row = row
+        @provider = provider
+        @messages = []
+
+        find!
+        validate!
+      end
+
+      def trainee
+        @trainees.first if @trainees.size == 1
+      end
+
+      def valid?
+        @messages.empty?
+      end
+
+      attr_reader :messages
+
+    private
+
+      attr_reader :row, :provider
+
+      def find!
+        trainees_by_trn!
+        trainees_by_hesa_id!
+        trainees_by_provider_trainee_id!
+      end
+
+      def validate!
+        @messages << "Multiple trainees found via #{found_with}" if @trainees.size > 1
+        @messages << "No trainee could be matched" if @trainees.empty?
```

Comment:
From the spreadsheet. are we ok to absorb the error "row has trainee data but no identifiers to match on" (i.e. no trn/provider_trainee_id/hesa_id) into this "No trainee could be matched" error? maybe one for @johndoates

Reply:
I think we'd want something along the lines of 'You've provided trainee data but no identifiers to match a trainee on'